mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -182,16 +182,25 @@ impl RpcCommonOps for TendermintCoin {
         let mut client_impl = self.client.0.lock().await;
         // try to find first live client
         for (i, client) in client_impl.rpc_clients.clone().into_iter().enumerate() {
-            if client
-                .perform(HealthRequest)
-                .timeout(Duration::from_secs(3))
-                .await
-                .is_ok()
-            {
-                // Bring the live client to the front of rpc_clients
-                client_impl.rpc_clients.rotate_left(i);
-                return Ok(client);
-            }
+            match client.perform(HealthRequest).timeout(Duration::from_secs(5)).await {
+                Ok(Ok(_)) => {
+                    // Bring the live client to the front of rpc_clients
+                    client_impl.rpc_clients.rotate_left(i);
+                    return Ok(client);
+                },
+                Ok(Err(rpc_error)) => {
+                    debug!(
+                        "Could not perform healthcheck on: {:?}. Error: {:?}",
```

Comment:
Better to display the error than debug
```suggestion
                        "Could not perform healthcheck on: {:?}. Error: {}",
```

Reply:
It's a debug log, so isn't it better to include every detail we have?

Reply:
For the timeout error, `future has timed out` will only be displayed when using `{}`. `rpc_error` also has some good error messages implemented.

Reply:
Oh I see, it has private fields lol. Thanks, good catch!